iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/pipeconsensus/PipeConsensusSyncConnector.java

```diff
@@ -0,0 +1,529 @@
+          PipeConsensusSyncConnector.class.getName())) {
+        return;
+      }
+      doTransfer(pipeInsertNodeTabletInsertionEvent);
+    } finally {
+      pipeInsertNodeTabletInsertionEvent.decreaseReferenceCount(
+          PipeConsensusSyncConnector.class.getName(), false);
+    }
+  }
+
+  private void doTransfer(PipeInsertNodeTabletInsertionEvent pipeInsertNodeTabletInsertionEvent)
+      throws PipeException {
+    final InsertNode insertNode;
+    final ProgressIndex progressIndex;
+    final TPipeConsensusTransferResp resp;
+    TCommitId tCommitId =
+        new TCommitId(
+            pipeInsertNodeTabletInsertionEvent.getCommitId(),
+            pipeInsertNodeTabletInsertionEvent.getRebootTimes());
+    TConsensusGroupId tConsensusGroupId =
+        new TConsensusGroupId(TConsensusGroupType.DataRegion, Integer.parseInt(consensusGroupId));
+
+    try (final SyncPipeConsensusServiceClient syncPipeConsensusServiceClient =
+        syncRetryAndHandshakeClientManager.borrowClient(getFollowerUrl())) {
+      insertNode = pipeInsertNodeTabletInsertionEvent.getInsertNodeViaCacheIfPossible();
+      progressIndex = pipeInsertNodeTabletInsertionEvent.getProgressIndex();
+
+      if (insertNode != null) {
+        resp =
+            syncPipeConsensusServiceClient.pipeConsensusTransfer(
+                PipeConsensusTabletInsertNodeReq.toTPipeConsensusTransferReq(
+                    insertNode, tCommitId, tConsensusGroupId, progressIndex, thisDataNodeId));
+      } else {
+        resp =
+            syncPipeConsensusServiceClient.pipeConsensusTransfer(
+                PipeConsensusTabletBinaryReq.toTPipeConsensusTransferReq(
+                    pipeInsertNodeTabletInsertionEvent.getByteBuffer(),
+                    tCommitId,
+                    tConsensusGroupId,
+                    progressIndex,
+                    thisDataNodeId));
+      }
+    } catch (Exception e) {
+      throw new PipeConnectionException(
+          String.format(
+              PIPE_CONSENSUS_SYNC_CONNECTION_FAILED_FORMAT,
+              getFollowerUrl().getIp(),
+              getFollowerUrl().getPort(),
+              e.getMessage(),
+              TABLET_INSERTION_NODE_SCENARIO),
+          e);
+    }
+
+    final TSStatus status = resp.getStatus();
+    // Only handle the failed statuses to avoid string format performance overhead
+    if (status.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()
+        && status.getCode() != TSStatusCode.REDIRECTION_RECOMMEND.getStatusCode()) {
+      receiverStatusHandler.handle(
+          status,
+          String.format(
+              "PipeConsensus transfer PipeInsertNodeTabletInsertionEvent %s error, result status %s",
+              pipeInsertNodeTabletInsertionEvent, status),
+          pipeInsertNodeTabletInsertionEvent.toString());
+    }
+  }
+
+  private void doTransferWrapper(final PipeRawTabletInsertionEvent pipeRawTabletInsertionEvent)
```

Comment:
If we do not support  "rawTablet" in receiver, maybe not supporting raw tablet in sync/async connector together is better.

Reply:
fixed